modules/perception/camera/app/cipv_camera.cc

```diff
@@ -0,0 +1,967 @@
+    if (debug_level_ >= 1) {
+      AINFO << "distance from end to right lane(" << distance
+            << " m) is too long";
+    }
+    return false;
+  }
+  if (distance_end_point_to_left_lane > MAX_DIST_OBJECT_TO_LANE_METER) {
+    if (debug_level_ >= 1) {
+      AINFO << "distance from end to left lane(" << distance
+            << " m) is too long";
+    }
+    return false;
+  }
+  distance = static_cast<float>(fabs(distance_start_point_to_right_lane -
+                  distance_end_point_to_right_lane));
+  if (distance > MAX_VEHICLE_WIDTH_METER) {
+    if (debug_level_ >= 1) {
+      AINFO << "width of vehicle (" << distance << " m) is too long";
+    }
+    return false;
+  }
+
+  distance = static_cast<float>(
+    fabs(distance_end_point_to_left_lane - distance_start_point_to_left_lane));
+  if (distance > MAX_VEHICLE_WIDTH_METER) {
+    if (debug_level_ >= 1) {
+      AINFO << "width of vehicle (" << distance << " m) is too long";
+    }
+    return false;
+  }
+  // put more conditions here if required.
+
+  // AINFO << "Distances are sane!";
+
+  return true;
+}
+
+// Check if a point is left of a line segment
+bool Cipv::IsPointLeftOfLine(const Point2Df &point,
+                             const Point2Df &line_seg_start_point,
+                             const Point2Df &line_seg_end_point) {
+  float cross_product = ((line_seg_end_point[0] - line_seg_start_point[0]) *
+                         (point[1] - line_seg_start_point[1])) -
+                        ((line_seg_end_point[1] - line_seg_start_point[1]) *
+                         (point[0] - line_seg_start_point[0]));
+
+  if (cross_product > 0.0f) {
+    if (debug_level_ >= 2) {
+      AINFO << "point (" << point[0] << ", " << point[1]
+            << ") is left of line_segment (" << line_seg_start_point[0] << ", "
+            << line_seg_start_point[1] << ")->(" << line_seg_end_point[0]
+            << ", " << line_seg_end_point[1]
+            << "), cross_product: " << cross_product;
+    }
+    return true;
+  } else {
+    if (debug_level_ >= 2) {
+      AINFO << "point (" << point[0] << ", " << point[1]
+            << ") is right of line_segment (" << line_seg_start_point[0] << ", "
+            << line_seg_start_point[1] << ")->(" << line_seg_end_point[0]
+            << ", " << line_seg_end_point[1]
+            << "), cross_product: " << cross_product;
+    }
+    return false;
+  }
+}
+// Check if the object is in the lane in image space
+bool Cipv::IsObjectInTheLaneImage(const std::shared_ptr<base::Object> &object,
+                                  const EgoLane &egolane_image) {
+  return true;
+}
+
+// Check if the object is in the lane in ego-ground space
+//  |           |
+//  | *------*  |
+//  |         *-+-----*
+//  |           |  *--------* <- closest edge of object
+// *+------*    |
+//  |           |
+// l_lane     r_lane
+bool Cipv::IsObjectInTheLaneGround(const std::shared_ptr<base::Object> &object,
```

Comment:
if possible, pls add unit test for this class

Reply:
I will add Unit test in the next PR.

Reply:
I added unit test in this PR.